design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,231 @@
+This has caused confusion in the past, so it's important to clear that up here.
+
+## Allowed Resource Access
+
+As previously stated, Crossplane is not intended to be a typical/full Kubernetes cluster to run arbitrary workloads.
+Once installed into the control plane, stacks will mostly target external resources and a known set of CRDs that they own and depend on.
+Therefore, we can lock down the allowed permissions for stacks to a subset of all available permissions in the Kubernetes API.
+
+The `rbac.yaml` file that allows a stack to specify the permissions it requires will be removed and no longer supported.
+Instead, the permissions for a stack can be considered **implicit**.
+The following permissions will be granted to a stack when it is installed:
+
+* **Core Kubernetes API kinds:** Simple configuration and monitoring related types such as `ConfigMap`, `Secrets` and `Events`.
+There should be no need for a stack to create a Deployment, Pod, Volume, or other types like that.
+* **Owned CRDs:** All CRDs that the stack owns and are being installed by the stack.
+* **Dependent CRDs:** All CRDs that the stack directly depends on and will consume to perform its primary operations (e.g. `MySQLInstance` claim).
+
+The Stack Manager will create the necessary service account, role, and binding that includes these permissions and serves as the runtime security context that the Stack will operate in.
+This is described in greater details in the sections below.
+
+In the future, other security modes could be enabled such as a more explicit model, and the allowed set of permissions could be expanded, but we are intentionally starting now with a small scope.
+
+## Cluster Stacks
+
+Cluster Stacks can be thought of as a higher level of privilege than other types of stacks, similar to how kernel modules work in Linux.
+The controllers contained in a cluster stack will usually reconcile resources that are external to the cluster, for example cloud provider managed services.
+These controllers will also likely be written as full fledged controller-runtime based implementations, but that is not a requirement.
+
+Let's look at a specific example of the GCP cluster stack to understand the security and isolation model more thoroughly.
+First, the user will create an instance of a `ClusterStackInstall` that specifies which cluster stack they want to install.
+Then, when the Stack Manager installs the GCP stack, a few things will happen:
+
+* All CRDs related to GCP infrastructure and services will be installed to the cluster, such as `CloudSQLInstance`, `CloudSQLInstanceClass`, `GKECluster`, `GKEClusterClass`, `Network`, `Subnetwork`, etc.
+* A `ClusterRole` will be created at the cluster level that includes the RBAC rules for the [allowed core Kubernetes kinds](#allowed-resource-access), the stack's owned CRDs, and its dependent CRDs.
```

Comment:
`dependent` is ambiguous here; I have heard it used to mean the opposite relationship too. I would suggest using some version of "dependency" or "consumer" instead of "dependent" or "dependee"; `CRDs the stack directly depends on` would work, and `consumed CRDs` would also work.

Reply:
Sure that's a good point, I think `dependent` is actually being used incorrectly here.  Antecedent would be the correct term I think, but that's pretty rarely used.  I'll update usage of `dependent` to be more explicit and clear, for example "the CRDs that the stack depends on".